### PR TITLE
Fix delete db scenario

### PR DESCRIPTION
**Pull request by maurerle**
Another attempt to fix the issue in 
https://github.com/assume-framework/assume/issues/512#issuecomment-2531064464

the first two commits are valid anyway, the other two commits are used to reproduce the mentioned issue (unsuccessfully on my side though)

The first two commits do:
* improve handling of assumeException (when the ovrwrite strategies question in learning is answered with a "n")
* do not access self.db before initializing the eninge in `on_ready`

Eventually this helps

### CODE REVIEW

File: examples/examples.py [hunk 1]
```diff
@@ -107,7 +107,7 @@
     data_format = "local_db"  # "local_db" or "timescale"
 
     # select the example to run from the available examples above
-    example = "small"
+    example = "small_learning_1"
```
Comment (nick-harder) [inline examples/examples.py:110]: this should not change?

File: assume/scenario/loader_csv.py [hunk 2]
```diff
@@ -728,7 +728,7 @@ def load_scenario_folder(
     study_case: str,
     perform_evaluation: bool = False,
     terminate_learning: bool = False,
-    episode: int = 1,
+    episode: int = 0,
     eval_episode: int = 1,
```
Comment (nick-harder) [inline assume/scenario/loader_csv.py:732]: should eval_episode be 0 as well?

**Review by nick-harder — CHANGES_REQUESTED** [review]

File: examples/examples.py (hunk 1 above)
Comment (maurerle) [reply examples/examples.py:110]: > the other two commits are used to reproduce the mentioned issue

Yes, I will revert those, once we know whats up with #512

File: assume/scenario/loader_csv.py (hunk 2 above)
Comment (maurerle) [reply assume/scenario/loader_csv.py:732]: I didn't look into this change really, just applied it by recommendation in #512 but it did not help me reproduce an issue

**Comment by nick-harder** [comment]
this also closes #517

**Review by nick-harder — APPROVED** [review]

**Pull request merged**